sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -726,22 +746,29 @@ public String[] fieldNames() {
      * is not provided as updating column default values does not need to back-fill existing data.
      * Empty string means dropping the column default value.
      */
-    public String newDefaultValue() { return newDefaultValue; }
+    public String newDefaultValue() {
+      return newCurrentDefault == null ? "" : newCurrentDefault.getSql();
+    }
+
+    /**
+     * Returns the column default value as {@link DefaultValue} with a
+     * {@link org.apache.spark.sql.connector.expressions.Expression}.  The default value literal
+     * is not provided as updating column default values does not need to back-fill existing data.
+     * Null means dropping the column default value.
+     */
+    public DefaultValue newCurrentDefault() { return newCurrentDefault; }
 
     @Override
     public boolean equals(Object o) {
```

Comment:
did you generate `equals` and `hashCode` using IDEA?

Reply:
yes, whats the recommendation?

Reply:
ah sorry i used wrong template in Intellij